tests/test_kafka.py

```diff
@@ -0,0 +1,181 @@
+
+    def kafka_broker_dns():
+        cmd = "dcos {package_name} --name={service_name} endpoints broker".format(
+            package_name=KAFKA_PACKAGE_NAME, service_name=KAFKA_SERVICE_NAME)
+        try:
+            stdout = subprocess.check_output(cmd, shell=True).decode('utf-8')
+        except Exception as e:
+            raise e("Failed to get broker endpoints")
+
+        return json.loads(stdout)["dns"][0]
+
+    kerberos_flag = "true" if KERBERIZED_KAFKA else "false"  # flag for using kerberized kafka given to app
+    stop_count = "48"  # some reasonable number
+    broker_dns = kafka_broker_dns()
+    topic = "top1"
+
+    big_file, big_file_url = "/mnt/mesos/sandbox/big.txt", "http://norvig.com/big.txt"
+
+    # arguments to the application
+    producer_args = " ".join([broker_dns, big_file, topic, kerberos_flag])
+
+    uris = "spark.mesos.uris=http://norvig.com/big.txt"
+    if KERBERIZED_KAFKA:
+        jaas_path = os.path.join(THIS_DIR, "resources", "spark-kafka-client-jaas.conf")
+        s3.upload_file(jaas_path)
+        jaas_uri = s3.s3_http_url("spark-kafka-client-jaas.conf")
+        uris += ",{}".format(jaas_uri)
+
+    common_args = [
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0",
+        "--conf", uris
+    ]
+
+    kerberos_args = [
+        "--conf", "spark.mesos.driver.secret.names=__dcos_base64___keytab",
+        "--conf", "spark.mesos.driver.secret.filenames=kafka-client.keytab",
+        "--conf", "spark.mesos.executor.secret.names=__dcos_base64___keytab",
+        "--conf", "spark.mesos.executor.secret.filenames=kafka-client.keytab",
+        "--conf", "spark.mesos.task.labels=DCOS_SPACE:/spark",
+        "--conf", "spark.executorEnv.KRB5_CONFIG_BASE64={}".format(KAFKA_KRB5),
+        "--conf", "spark.mesos.driverEnv.KRB5_CONFIG_BASE64={}".format(KAFKA_KRB5),
+        "--conf", "spark.driver.extraJavaOptions=-Djava.security.auth.login.config="
+                  "/mnt/mesos/sandbox/spark-kafka-client-jaas.conf",
+        "--conf", "spark.executor.extraJavaOptions="
+                  "-Djava.security.auth.login.config=/mnt/mesos/sandbox/spark-kafka-client-jaas.conf",
+    ]
+
+    producer_config = ["--conf", "spark.cores.max=2", "--conf", "spark.executor.cores=2",
+                       "--class", "KafkaProducer"] + common_args
+
+    if KERBERIZED_KAFKA:
+        producer_config += kerberos_args
+
+    producer_id = utils.submit_job(app_url=utils._scala_test_jar_url(),
+                                   app_args=producer_args,
+                                   app_name="/spark",
+                                   args=producer_config)
+
+    shakedown.wait_for(lambda: producer_launched(), ignore_exceptions=False, timeout_seconds=600)
+    shakedown.wait_for(lambda: producer_started(), ignore_exceptions=False, timeout_seconds=600)
```

Comment:
Nit: you could use this method instead: https://github.com/mesosphere/spark-build/blob/master/tests/utils.py#L196